with-camera/App.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react';
+import { Text, View, TouchableOpacity, ImageBackground, Platform } from 'react-native';
+import { Camera } from 'expo-camera';
+
+export default function App() {
+  const [hasPermission, setHasPermission] = useState(null);
+  const [type, setType] = useState(Camera.Constants.Type.back);
+  const [lastPhotoURI, setLastPhotoURI] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS == "web") { 
+        setHasPermission(true);
+        return; 
+      }
+      const { status } = await Camera.requestPermissionsAsync();
+      setHasPermission(status === 'granted');
+    })();
+  }, []);
+
+  if (lastPhotoURI !== null) {
+    return (
+      <ImageBackground 
+        source={{uri: lastPhotoURI}}
+        style={{
+          flex: 1,
+          backgroundColor: 'transparent',
+          flexDirection: 'row',
+          justifyContent: "center"
+        }}>
+        <TouchableOpacity
+            style={{
+              flex: 0.2,
+              alignSelf: 'flex-end',
+              alignItems: 'center',
+              justifyContent: "center",
+              backgroundColor: "#666",
+              marginBottom: 40, 
+              marginLeft: 20,
+            }}
+            onPress={() => {
+              setLastPhotoURI(null);
+            }}>
+            <Text style={{ fontSize: 30, padding: 10, color: 'white' }}>❌</Text>
+          </TouchableOpacity>
+      </ImageBackground>
+    );
+  }
+
+  if (hasPermission === null) {
+    return <View />;
+  }
+  if (hasPermission === false) {
+    return <Text>No access to camera</Text>;
+  }
+
+  let cameraRef;
+  return (
+    <View style={{ flex: 1 }}>
+      <Camera 
+        style={{ flex: 1 }}
+        type={type}
+        ref={ref => {
+          cameraRef = ref;
+        }}
```

Comment:
```suggestion
        ref={cameraRef}
```

Use the hook `React.useRef(null)` instead